distributed/deploy/ssh.py

```diff
@@ -204,10 +218,37 @@ async def start(self):
             logger.info(line.strip())
             if "Scheduler at" in line:
                 self.address = line.split("Scheduler at:")[1].strip()
+                if self.forward_scheduler_port:
+                    port = self.address.split(":")[-1]
+                    self.port_forwards.append(
+                        await self.connection.forward_remote_port(
+                            "localhost",
+                            int(self.forward_scheduler_port),
+                            "localhost",
+                            int(port),
+                        )
+                    )
+                    self.address = "tcp://localhost:{}".format(
+                        self.forward_scheduler_port
+                    )
```

Comment:
What if we're using a different protocol, like tls?

Reply:
Ah whoops I meant to fix that.

Reply:
Resolved, now auto detects in the same way it detects the port on the remote end.